src/Microsoft.Azure.WebJobs.Host.Storage/RuntimeConfigurationExtensions.cs

```diff
@@ -0,0 +1,101 @@
+using Microsoft.Azure.WebJobs.Host.Loggers;
+using Microsoft.Azure.WebJobs.Host.Configuration;
+using WebJobs.Host.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace Microsoft.Extensions.Hosting
+{
+    public static class RuntimeConfigurationExtensions
+    {
+        // WebJobs v1 Classic logging. Needed for dashboard.         
+        // $$$ Update title? 
+        public static IHostBuilder AddWebJobsLogging(this IHostBuilder builder)
+        {
+            return builder
+               .ConfigureServices((context, services) =>
+               {
+                   services.AddWebJobsLogging();
+               });
+        }
+
+        // WebJobs v1 Classic logging. Needed for dashboard.         
+        public static IServiceCollection AddWebJobsLogging(this IServiceCollection services)
+        {
+            // Logging related services (lots of them...)
+            services.TryAddSingleton<LoggerProviderFactory>();
+                        
+            services.TryAddSingleton<IFunctionOutputLoggerProvider>(p => p.GetRequiredService<LoggerProviderFactory>().GetLoggerProvider<IFunctionOutputLoggerProvider>());
+            services.TryAddSingleton<IFunctionOutputLogger>(p => p.GetRequiredService<IFunctionOutputLoggerProvider>().GetAsync(CancellationToken.None).GetAwaiter().GetResult());
+
+            services.TryAddSingleton<IFunctionInstanceLoggerProvider>(p => p.GetRequiredService<LoggerProviderFactory>().GetLoggerProvider<IFunctionInstanceLoggerProvider>());
+            services.TryAddSingleton<IFunctionInstanceLogger>(p => p.GetRequiredService<IFunctionInstanceLoggerProvider>().GetAsync(CancellationToken.None).GetAwaiter().GetResult());
+
+            services.TryAddSingleton<IHostInstanceLoggerProvider>(p => p.GetRequiredService<LoggerProviderFactory>().GetLoggerProvider<IHostInstanceLoggerProvider>());
+            services.TryAddSingleton<IHostInstanceLogger>(p => p.GetRequiredService<IHostInstanceLoggerProvider>().GetAsync(CancellationToken.None).GetAwaiter().GetResult());
+
+            return services;
+        }
+
+        // Make the Runtime itself use storage for its internal operations. 
+        // Uses v1 app settings, via a LegacyConfigSetup object. 
+        public static IHostBuilder AddStorageForRuntimeInternals(this IHostBuilder builder)
+        {
+            return builder
+               .ConfigureServices((context, services) =>
+               {
+                   // Add runtime services that depend on storage.
+                   services.TryAddSingleton<BlobManagerXStorageAccountProvider>(); 
+                   services.AddSingleton<IDistributedLockManager>(provider => Create(provider));
+                                      
+                   services.TryAddSingleton<IHostIdProvider, DynamicHostIdProvider>();
+
+                   services.TryAddEnumerable(ServiceDescriptor.Transient<IConfigureOptions<LegacyConfig>, LegacyConfigSetup>());
+
+                   // $$$
```

Comment:
There are 111 instances of this $$$ temp marker in this branch currently, presumably marking things that need to be fixed up. Can these be addressed/removed?